### PR TITLE
Remove DiscoveryNodes.getSmallestNonClientNodeVersion method

### PR DESCRIPTION
**Pull request by thecoop**
This is technically a behaviour change - in mixed node clusters, when there are still nodes on the old version that don't store data, the old code will run the operations, whereas this will wait until every node is upgraded.

I don't think this is an issue, as these are operations that are not time sensitive.

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticmachine update branch

**Comment by mosche** [comment]
For SystemIndexMapping updates the new behavior makes more sense to me, but I might be missing some context: Could there be situations where we need the intermediate update to/by the oldest node version?

**Comment by thecoop** [comment]
What do you mean by 'intermediate update'?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/TransportPostStartTrialAction.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ protected void masterOperation(
         ClusterState state,
         ActionListener<PostStartTrialResponse> listener
     ) throws Exception {
-        if (state.nodes().getMaxNodeVersion().after(state.nodes().getSmallestNonClientNodeVersion())) {
+        if (state.nodes().isMixedVersionCluster()) {
```
Comment (thecoop) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/license/TransportPostStartTrialAction.java:58]: @gwbrown Is this ok?

**Comment by mosche** [comment]
> What do you mean by 'intermediate update'?

Actually, ignore that, I misunderstood. LGTM, but prob best to get confirmation from Athena

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/TransportPostStartTrialAction.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/TransportPostStartTrialAction.java:58]: Yeah, that's probably better.

**Review by mosche — APPROVED** [review]

**Pull request merged**